checker/src/main/java/org/checkerframework/checker/initialization/InitializationAnnotatedTypeFactory.java

```diff
@@ -847,19 +1166,21 @@ public Void visitMemberSelect(
             }
             return super.visitMemberSelect(tree, annotatedTypeMirror);
         }
+
+        /* The result of a binary or unary operator is always @Initialized. */
```

Comment:
Is this assertion true for a type system that handles primitive types?
I think this code is based on the old "NullnessPropagationTreeAnnotator" code that said: "Nullness doesn't call propagation on binary and unary because the result is always @Initialized (the default qualifier)."
In a hypothetical system that handles division by zero, does that assertion still hold?